mautic/mautic
# [UX] Tip when creating new contact manually

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description

### 10k lines of code because I moved variables between files

users manually entering contacts might not realize there's a bulk import feature available. this pr adds a tip to the new contact creation interface, informing users about the bulk import option

![image](https://github.com/user-attachments/assets/48d13fe3-5a7c-4284-833f-ca97fa3034e6)

one of the main features here is the possibility to dimiss a tip FOREVER

but you can also see dismissed things again if you reset in the account config

![image](https://github.com/user-attachments/assets/e0c9fe17-fc92-41d0-a451-d99c3a2bcf2a)

other changes (technical stuff |if you're a non-technical tester, please ignore ):

- moved variables from tokens.less to variables.less
- moved some other parts of the code between files to help devs get organized (you'll see no changes caused by this)
- Added margin-left to dropdown-toggle, so it gets separate from the other buttons
- Recovered old panel styles like the panel footer background
- Improved the reliability of border-radius varibles
- Updated parts of the code that were using old variable names and weren't working
- Removed import rules from files that didn't need them
- Made a small fix to the **Add note** button border radius when viewing a contact
- Fixed text appearing with no spaces between words when searching for things in the campaign builder @escopecz 
- Dropdowns are now translucent (semi transparent background color with blur -- it can be disabled in account settings)
- Improved the general transparency from the --layer-translucent variable across all default themes
- Improved popover styles to make the header translucent (also can be disabled in account settings)
- Created a JS function that stores dismissed tips in the localStorage and append hiding rules to ensure you won't see them again until reset
- The JS function ensures that you'll never have glitches when reloading the page, on ajaxComplete event, documentReady and absolutely any other situation
- Improved the "No information" template to rely less on alert-info styles (we don't want them to be alerts)
- Added new notification/alerts styles, now they have background
- Spacing improvements to the headings in the account settings
- Fixed many small UI bugs
- Started using default Bootstrap logic to make everyone's life easier (border-radius, mainly)
- Created an inline_notification Twig template, so now devs can add things without the design hassle (with automatic accessibility! docs soon)
- and many more...







---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Go to Contacts > Create new and ONLY see the info
3. Save & Close the contact
4. Then Edit it to see the message is not there
5. Go to the Contacts page > Create new
6. Dismiss the message clicking on the X
7. Reload the page (it won't be there anymore)
8. Open the user dropdown on the navbar
9. Click Account 
10. On the account page, click Restart dismissed tips
11. Repeat 5-7
